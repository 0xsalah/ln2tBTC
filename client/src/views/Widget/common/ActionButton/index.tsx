import React from "react";

interface ActionButtonProps {
  text: string;
  className?: string;
  type?: "primary" | "secondary" | "loading";
  disabled?: boolean;
  onClick?: () => void;
}

const ActionButton: React.FC<ActionButtonProps> = (props) => {
  const {
    text,
    className = "",
    type = "primary",
    disabled = false,
    onClick = () => null,
  } = props;
  return (
    <div className={className}>
      <button
        onClick={onClick}
        className={`btn btn--full btn--${type}`}
        disabled={type === "loading" || disabled}
      >
        <span>{text}</span>
        {type === "loading" && <i className="icon icon-loader"></i>}
      </button>
    </div>
  );
};

export default ActionButton;
