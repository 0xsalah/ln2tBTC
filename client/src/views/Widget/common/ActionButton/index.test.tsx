import React from "react";
import { render, screen } from "@testing-library/react";
import ActionButton from ".";

test("snapshot of button with default attributes", () => {
  const component = render(<ActionButton text="Test" />);
  expect(component.baseElement).toMatchSnapshot();
});

test("snapshot of button disabled", () => {
  const component = render(<ActionButton text="Test" disabled />);
  expect(component.baseElement).toMatchSnapshot();
});

test("snapshot of button type primary", () => {
  const component = render(<ActionButton text="Test" type="primary" />);
  expect(component.baseElement).toMatchSnapshot();
});

test("snapshot of button type secondary", () => {
  const component = render(<ActionButton text="Test" type="secondary" />);
  expect(component.baseElement).toMatchSnapshot();
});

test("snapshot of button type loading", () => {
  const component = render(<ActionButton text="Test" type="loading" />);
  expect(component.baseElement).toMatchSnapshot();
});

test("click event default type button", () => {
  const mockCallBack = jest.fn();

  const component = render(<ActionButton text="Test" onClick={mockCallBack} />);
  const button = screen.getByText("Test").closest("button");
  if (button) button.click();
  expect(mockCallBack.mock.calls.length).toEqual(1);
});

test("click event primary type button", () => {
  const mockCallBack = jest.fn();

  const component = render(
    <ActionButton text="Test" onClick={mockCallBack} type="primary" />
  );
  const button = screen.getByText("Test").closest("button");
  if (button) button.click();
  expect(mockCallBack.mock.calls.length).toEqual(1);
});

test("click event secondary type button", () => {
  const mockCallBack = jest.fn();

  const component = render(
    <ActionButton text="Test" onClick={mockCallBack} type="secondary" />
  );
  const button = screen.getByText("Test").closest("button");
  if (button) button.click();
  expect(mockCallBack.mock.calls.length).toEqual(1);
});

test("click event disabled type button", () => {
  const mockCallBack = jest.fn();

  const component = render(
    <ActionButton text="Test" onClick={mockCallBack} disabled />
  );
  const button = screen.getByText("Test").closest("button");
  if (button) button.click();
  expect(mockCallBack.mock.calls.length).toEqual(0);
});

test("click event loading type button", () => {
  const mockCallBack = jest.fn();

  const component = render(
    <ActionButton text="Test" onClick={mockCallBack} disabled />
  );
  const button = screen.getByText("Test").closest("button");
  if (button) button.click();
  expect(mockCallBack.mock.calls.length).toEqual(0);
});
