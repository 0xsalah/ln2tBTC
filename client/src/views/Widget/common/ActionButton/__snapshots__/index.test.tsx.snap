// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of button disabled 1`] = `
<body>
  <div>
    <div
      class=""
    >
      <button
        class="btn btn--full btn--primary"
        disabled=""
      >
        <span>
          Test
        </span>
      </button>
    </div>
  </div>
</body>
`;

exports[`snapshot of button type loading 1`] = `
<body>
  <div>
    <div
      class=""
    >
      <button
        class="btn btn--full btn--loading"
        disabled=""
      >
        <span>
          Test
        </span>
        <i
          class="icon icon-loader"
        />
      </button>
    </div>
  </div>
</body>
`;

exports[`snapshot of button type primary 1`] = `
<body>
  <div>
    <div
      class=""
    >
      <button
        class="btn btn--full btn--primary"
      >
        <span>
          Test
        </span>
      </button>
    </div>
  </div>
</body>
`;

exports[`snapshot of button type secondary 1`] = `
<body>
  <div>
    <div
      class=""
    >
      <button
        class="btn btn--full btn--secondary"
      >
        <span>
          Test
        </span>
      </button>
    </div>
  </div>
</body>
`;

exports[`snapshot of button with default attributes 1`] = `
<body>
  <div>
    <div
      class=""
    >
      <button
        class="btn btn--full btn--primary"
      >
        <span>
          Test
        </span>
      </button>
    </div>
  </div>
</body>
`;
